integration/hibernate-base/src/main/java/com/blazebit/persistence/integration/hibernate/base/HibernateJpaProvider.java

```diff
@@ -866,8 +868,14 @@ public void setCacheable(Query query) {
         boolean supportsNaturalIdAccessOptimization = false;
         AbstractEntityPersister entityPersister = getEntityPersister(owner);
         Type propertyType = entityPersister.getPropertyType(attributeName);
-        org.hibernate.type.EntityType entityType = (org.hibernate.type.EntityType) propertyType;
+
+        // For nested embedded id's, the nested id is mapped as a component type
+        if (propertyType instanceof ComponentType) {
```

Comment:
Shouldn't this be done in a recursive manner?

Reply:
The only way I have reached this path is with with nesting of `@Id` association properties using `@IdClass`, i.e. `someAlias.someIdAssocation.someOtherIdAssocation.compoundKeyPart`. The type for `someAlias.someIdAssocation` is `EntityType`, but for `someAlias.someIdAssocation.someOtherIdAssocation` it is `ComponentType` (even though I would somewhat expect an `EntityType here). I haven't reached this path in any other way. Because Embeddables are almost never Joinable I don't really see how this could ever happen. Do you have a concrete example in mind?

Reply:
Now that I think of it, it's fine!